include/solvers/gls_sharp_navier_stokes.h

```diff
@@ -60,165 +61,193 @@ class GLSSharpNavierStokesSolver : public GLSNavierStokesSolver<dim>
+   * This structure gives access to the position, velocity, force and other
+   * proprieties of each IB particle. All the variables defined for each of the
+   * particle are described in the class: IBparticle. see:
+   * \include\core\ib_particle.h
+   */
+  void
+  define_particles();
+
+  /**
+   * @brief
+   * Evaluate the forces applied on each of the IB particle.
+   */
+  void
+  force_on_ib();
+
+  /**
+   * @brief
+   * Modify the system matrix to impose IB condition using the sharp_edge
+   * approach. The detail of this approach are presented in : L. Barbeau, S.
+   * Étienne, C. Béguin & B. Blais, «Development of a high-order continuous
+   * Galerkin sharp-interface immersed boundary method and its application to
+   * incompressible flow problems,» Computers & Fluids, 2020, in press, ref.
+   * CAF-D-20-00773
+   */
+  void
+  sharp_edge();
+
+  /**
+   * @brief
+   * Write in a specifique file for each of the paticles its forces, velocity,
+   * position at each time step. LB - TODO refactor the output format of the
+   * file
+   */
+  void
+  write_force_ib();
+
+  /**
+   * @brief
+   * Integrate the particle velocity and position based on the forces and
+   * torques and applies the next value to the particle.
+   */
+  void
+  integrate_particles();
+
+  /**
+   * @brief
+   * Store the solution of the particles dynamics parameters for integration.
+   * Defines the table to store the history of each of the particles.
+   */
+  void
+  finish_time_step_particles();
+
+  /**
+   * @brief
+   * Evaluate the L2 error on the computational domain if an analytical solution
+   * is given. This function is slightly different from its standard GLS
+   * counterpart as the cells that are cut by an immersed boundary should not be
+   * taken into account in the error evaluation. See "computation domain"
+   * definition in: L. Barbeau, S. Étienne, C. Béguin & B. Blais, «Development
+   * of a high-order continuous Galerkin sharp-interface immersed boundary
+   * method and its application to incompressible flow problems,» Computers &
+   * Fluids, 2020, in press, ref. CAF-D-20-0077
+   */
+  double
+  calculate_L2_error_particles();
+
+
+
+  /**
+   * @brief
+   * Same function as its standard GLS counterpart but it used the error
+   * evaluation that takes into account the particle’s position.
+   */
+  virtual void
+  postprocess_fd(bool firstIter) override;
+
+  /**
+   * @brief
+   * Allow a refinement around each of the particles.
+   * The zone where the cells will be refined is defined by a ring in 2D and a
+   *shell in 3D. The outside and inside radius of the ring\shell is defined in
+   *relation to the diameter of the particle by the immersed boundaries
+   *parameter: "refine mesh inside radius factor" and "refine mesh outside
+   *radius factor". These factors multiply the radius of the particle to define
+   *the outside and inside radius of the ring\shell.
```

Comment:
Did you apply automatic indentation?

Reply:
yeah this is after the indentation function call